docs/GettingStarted.md

```diff
@@ -1,3 +1,269 @@
+However, we believe that we can help alleviate the amount of boilerplate and overhead by providing
+all the useful things for you!
+
+Likewise, you will see that the [PropTypes](/Component/PropTypes.html) object of Shasta contains both React's native prop types
+as well as immutablejs' prop types.
+
+Our component will need the store as a prop so let's change `src/index.js` to provide it:
+
+```js
+import DOM from 'react-dom';
+import React from 'react';
+import store from './core/store';
+import App from './App.js';
+
+const rootNode = <App store={store} />;
+DOM.render(rootNode, document.getElementById('root'));
+```
+
+## The store
+
+We will need to use the store in several places so let's create it.
+In `src/core/store.js`, use the `createStore` function to create and export our store.
+This way, we can import it and reference it as a singleton:
+
+```js
+import { createStore, createReducer } from 'shasta';
+import * as counter from '../reducers/counter';
+
+const localReducers = {
+  counter
+};
+
+export default createStore({
+  plugins: [],
+  reducers: [
+    createReducer(localReducers)
+  ]
+})
+```
+
+Shasta has a concept of plugins that you put directly into the store.
+Plugins will extend your application by providing reducers or middlewares, among other things.
+The idea is that you can add new capabilities to your application (the router, or access to web APIs) with a single line
+of code, thanks to plugins.
+
+## Our first reducer
+
+The store needs a counter reducer so let's write it at once:
+
+```js
+import { Map } from 'immutable';
+
+export const initialState = Map({ count: 1 });
+
+export const increment = (state, { payload = 1 }) =>
+  state.update('count', c => c + payload);
+
+export const decrement = (state, { payload = 1 }) =>
+  state.update('count', c => c - payload);
+
+export const reset = () => initialState;
+```
+
+Shasta helps you write straight-forward reducers as a set of functions, using immutable data structure.
+This makes them very easy to read.
+
+The best part is that we can access those reducer functions as actions anywhere in our application!
+
+## Actions
+
+Shasta believes that you should be able to access your reducers anywhere in your app.
+To do this, let's create `src/core/actions.js`:
+
+```js
+import { createActions, createReducerActions } from 'shasta';
+import store from './store'
+import * as counter from '../reducers/counter';
+
+const localReducers = {
+  counter
+};
+
+export default createActions({
+  ...createReducerActions(localReducers)
+}, store.dispatch)
+```
+
+This will effectively bind all our reducer functions to Redux's dispatch function.
+We can now use them in our components.
+
+## Using our reducer
+
+Finally, let's create our `Counter` component in `src/components/Counter.js`:
+
+```js
+import React from "react";
+import {connect, Component} from "shasta";
+import actions from "../core/actions";
+
+@connect({
+  count: 'counter.count'
+})
```

Comment:
@geowarin If something is really the true and best way to do it we should build it into the stack instead of documenting it, IMO. I think it does a pretty good job of forcing the user to structure things a certain way, but if something is still unclear we can add more strictness as needed.